components/keys/src/lib.rs

```diff
@@ -56,6 +60,9 @@ pub const SNAPSHOT_RAFT_STATE_SUFFIX: u8 = 0x04;
 // For region meta
 pub const REGION_STATE_SUFFIX: u8 = 0x01;
 
+// For peer meta
+pub const PEER_STATE_SUFFIX: u8 = 0x01;
```

Comment:
I don't think it's a good idea to add a new state just for one corner case, especially when it can be fixed by no extra states.

Reply:
I think this is not a corner case. On the contrary, it is the correct way to give each removed peer a separate key to record its status.

Reply:
>  especially when it can be fixed by no extra states.

@BusyJay Could you roughly describe the solution?

Reply:
It's quoted in the original issue that "adding the current peer into peer list when writing tombstone".

Reply:
> It's quoted in the original issue that "adding the current peer into peer list when writing tombstone".

This is to introduce a new concept just to solve a corner case, and only part of the problem can be solved. You also need to introduce some other mechanisms to resolve other case. For example, merge a region need the region info in the RegionLocalState to continue ticking for overlapped, so I need save both tombstone state and region info.

The most intuitive and easy to understand way is to record state and region info on the state of the peer, not on the state of the region. In this way, any related problems can be solved once and for all.

Reply:
> ...introduce a new concept just to solve a corner case...

What's the new concept? Using epoch and peer list to check for existence is existing flow.

> For example, merge a region need the region info in the RegionLocalState to continue ticking for overlapped, so I need save both tombstone state and region info.

Can you explain more? I don't see how adding the current peer into peer list can break this.

Reply:
> Using epoch and peer list to check for existence is existing flow.
Maybe I misunderstood what you said "add current peer into peer list", do you mean add the current peer into the region in RegionLocalState if it is uninitialized?

Reply:
Yes. More precisely, if a peer is about to be destroyed, and it's uninitialized, then add the peer to the region peer list without changing epoch. So when a stale heartbeat arrives, it will find the local state in tombstone with epoch set to 0. So it will check the peer list to find if there is a peer with larger or equal ID, then hit the added peer and return.

Reply:
Make sense, I will try to modify this PR.